test/classes/nilability/while-var.chpl

```diff
@@ -0,0 +1,40 @@
+
+// a concrete class
+class clist {
+  var elm: int;
+  var next: owned clist?;
+}
+
+// a generic class
+class glist {
+  var elm;
+  var next: owned glist(elm.type)?;
+}
+
+proc consume(name: string, in list) {
+  write("consume ", name, ":");
+  var curr = list: class?;
+  while const cur = curr {
```

Comment:
Ideally I'd like to see a version of this test with an explicit type declaration for this variable and a version of both of those cases with a `var` declaration instead.  I'm assuming it's important to provide the initial value, what happens when only a type is provided and an initial value is forgotten?  Or when neither occur and it would potentially be a mistrigger for split initialization?

Reply:
This is the only syntax currently supported - in both `if` and `while` cases - with the only allowed variation being `var` instead of `const`. In any case I am open to suggestions here.

Reply:
Oh, interesting.  Do we intend to support type declarations in the future?

Reply:
Potentially yes. For example, what if in the above code `list` is `shared` and I would like to preserve its shared-ness within the loop? To indicate that to the compiler, I suggest to have the user declare `cur` as a non-nilable shared.

Reply:
Okay, sweet!